.github/workflows/build.yml

```diff
@@ -0,0 +1,115 @@
+name: Build
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  ubuntu-gcc:
+    name: Ubuntu 18.04 / gcc
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v2
+    - name: Install build dependencies
+      run: |
+        sudo apt-get update && sudo apt-get install -y --no-install-recommends \
+          ccache \
+          libavformat-dev \
+          libchromaprint-dev \
+          libebur128-dev \
+          libfaad-dev \
+          libfftw3-dev \
+          libflac-dev \
+          libid3tag0-dev \
+          liblilv-dev \
+          libmad0-dev \
+          libmodplug-dev \
+          libmp3lame-dev \
+          libmp4v2-dev \
+          libopus-dev \
+          libopusfile-dev \
+          libportmidi-dev \
+          libprotobuf-dev \
+          libqt5opengl5-dev \
+          libqt5sql5-sqlite \
+          libqt5svg5-dev \
+          libqt5x11extras5-dev \
+          librubberband-dev \
+          libshout3-dev \
+          libsndfile1-dev \
+          libsoundtouch-dev \
+          libsqlite3-dev \
+          libtag1-dev \
+          libupower-glib-dev \
+          libusb-1.0-0-dev \
+          libwavpack-dev \
+          portaudio19-dev \
+          protobuf-compiler \
+          qt5-default \
+          qt5keychain-dev \
+          qtscript5-dev \
+    - name: Set up cmake
+      uses: jwlawson/actions-setup-cmake@v1.4
+      with:
+        cmake-version: '3.13.x'
+    - name: Create build directory
+      run: mkdir cmake_build
+    - name: Prepare ccache timestamp
+      id: ccache_cache_timestamp
+      shell: cmake -P {0}
+      run: |
+        string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+        message("::set-output name=timestamp::${current_date}")
+    - name: Set up ccache cache
+      uses: actions/cache@v2
+      env:
+        cache-name: ccache
+      with:
+        path: .cache/ccache
```

Comment:
This is incorrect.
```suggestion
        path: ~/.ccache
```

Reply:
Are you sure?

> **cache_dir (CCACHE_DIR)**
> This option specifies where ccache will keep its cached compiler outputs. It will only take effect if set in the system-wide configuration file or as an environment variable. **The default is $XDG_CACHE_HOME/ccache if XDG_CACHE_HOME is set, otherwise $HOME/.cache/ccache**. Exception: If the legacy directory $HOME/.ccache exists then that directory is the default.

https://ccache.dev/manual/4.0.html#_configuration_options

Reply:
Yes I am sure. I manually reran the build and the build time was the same despite the cache being loaded. I tested this with my branch too. If you run `ccache -p` on GitHub Actions it shows `~/.ccache` as the cache directory.

Reply:
Probably because the Ubuntu is old 🤷

Reply:
Why not add both paths, just in case? One location will always be empty.

Reply:
Fedora 33 with  CCache 3.7.11 also defaults to ~/.ccache. The new CCache 4 location ~/.cache/ccache is ignored and ~/.ccache is re-created implicitly.

Reply:
Version 4.0 has only been released a month ago. Version 3.7 will not disappear soon. By explicitly creating `~/.ccache` if it does not exist we only need to cache a single directory.

Reply:
We build with the oldest Ubuntu version we support so yes we will not be using ccache 4 on Linux for a long time. (I have just added ccache 4 to the macOS build environment.)